File: Original Configaration/Python/netnodecalc.py

def finder(a):
	return((int(a[1])-1, ord(a[0])-ord('A')))



def calc(i):
        if i%5 ==0:
                b = ((i-1)//5,5)
        else :
                b = (i//5,i%5)

        min = 0
        a =(b[0],b[1]-1) 
        c = (b[0]+1,b[1]) 
        d = (a[0]+1,a[1])



def calc_min(g,i):

	""" returns a node of an habitat thats is near to animal associated with """
	g = finder(g)
	if i%5 ==0:
		b = ((i-1)//5,5)
	else :
		b = (i//5,i%5)

	min = 100 ; marker = 0
	a =(b[0],b[1]-1)
	c = (b[0]+1,b[1])
	d = (a[0]+1,a[1])
	for x in [b,a,d,c]:
		temp = (x[0]-g[0])**2 + (x[1]-g[1])**2

		if temp < min:
			min = temp
			marker = x
	return(marker)

File: Original Configaration/Embedded C/Original_Traversal/trial/netnodecalc.c
#include<stdio.h>
struct cord
{

	int x  ; int y;

};


int calc_min(int b, int h)

{       struct cord a;
	a.x = b/10;
	a.y  = b%10;


//###############################################################################
//calculates the nearest co ordinate of habitat associated with animal co ordinate
//###############################################################################

	int d[4] ;
	d[1] = h;
	d[0] =  (  (d[1]/10)*10 ) + d[1]%10 -1;
	d[2] = (d[1]/10 +1 )*10  + d[1]%10;
	d[3] =  (d[0]/10 +1 )*10 + d[0]%10;

	int min = 100;
	int temp = 0;
	int temp2 ;
	int marker;
	for(int i = 0 ; i < 4 ;i ++)
	{
		temp = (d[i]/10 - a.x)*( d[i]/10 - a.x) + (d[i]%10 -a.y)*(d[i]%10 -a.y);
		if( temp < min)
	                     {    min=temp; marker=i;       }


	}

	return(d[marker]);


}
